### PR TITLE
Day 04 - Add Transaction Routes/Tests, Enhance Category Routes

### PR DESCRIPTION
**Pull request by kishanpatel789**
## Change Log
- Complete API routes and corresponding tests for Transaction PATCH and DELETE
- Add checks for Category POST/PATCH to prevent duplicate categories with the same name
  - Add tests to catch attempts to make duplicate Category name via API
- Update status code response from POST API routes to be 201 (entity created)
- Rename some test functions to be consistent with API route functions

## TODO
- [ ] Add validation to data models (eg. strip whitespace, enforce min and max string lengths, etc)
- [ ] Add feature to search transactions in API (to support reporting in frontend UI)
- [ ] Add pagination to GET transactions endpoint 
- [ ] Set up Postgres database in Docker to replace current SQLite database

### CODE REVIEW

File: src/routers/categories.py [hunk 1]
```diff
@@ -16,9 +16,25 @@
 )
 
 
-@router.post("/", response_model=CategoryRead)
+def check_for_existing_category(session: SessionDep, category_name: str) -> None:
+    """Check if a category with the given name already exists."""
+
+    db_category_existing = session.exec(
+        select(Category).where(Category.name == category_name)
```
Comment (bbelderbos) [inline src/routers/categories.py:23]: we don't need case insensitivity here? so Food and food can co-exist?

File: tests/test_api_categories.py [hunk 2]
```diff
@@ -77,6 +84,15 @@ def test_update_category(client: TestClient, add_category):
     assert data["budget"] == "200.00"
 
 
+def test_update_category_conflicting_name(
+    client: TestClient, add_category, add_another_category
+):
+    payload = {"name": "Utilities", "budget": None}
```
Comment (bbelderbos) [inline tests/test_api_categories.py:90]: What if you want to update the budget on an existing category, would that work? I hope this validation is not too aggressive?

File: src/routers/categories.py (hunk 1 above)
Comment (bbelderbos) [reply src/routers/categories.py:23]: Awesome, that sounds like an elegant solution.

File: tests/test_api_categories.py (hunk 2 above)
Comment (bbelderbos) [reply tests/test_api_categories.py:90]: Yep, thanks for confirming, I think this works indeed because you check for a name change first. Good to confirm this with a test though. 

Could also add a helper for readability:

```
def _category_name_changed(category_data: dict, db_category: Category) -> bool:
    return category_data.get("name") and category_data["name"] != db_category.name

Comment (kishanpatel789) [reply tests/test_api_categories.py:90]: Good idea! I'm adding the helper function and test in the next PR. Thanks for the tip!
